pkg/lint/rules/deprecations_test.go

```diff
@@ -27,7 +30,11 @@ func TestValidateNoDeprecations(t *testing.T) {
 	if err == nil {
 		t.Fatal("Expected deprecated extension to be flagged")
 	}
-	depErr := err.(deprecatedAPIError)
+	depErr := func() deprecatedAPIError {
+		var target deprecatedAPIError
+		_ = errors.As(err, &target)
```

Comment:
this is functionally different?

`depErr := err.(deprecatedAPIError)` will panic if the cast fails. `_ = errors.As(err, &target)` will ignore the conversion failure?

Reply:
(and similar code below)